src/app/CommandSender.h

```diff
@@ -36,15 +38,71 @@
 #include <system/SystemPacketBuffer.h>
 
 #include <app/Command.h>
+#include <app/MessageDef/CommandPath.h>
+#include <app/MessageDef/StatusElement.h>
 
 #define COMMON_STATUS_SUCCESS 0
 
 namespace chip {
 namespace app {
 
-class CommandSender : public Command, public Messaging::ExchangeDelegate
+class CommandSender final : public Command, public Messaging::ExchangeDelegate
 {
 public:
+    class Callback
+    {
+    public:
+        virtual ~Callback() = default;
+
+        /**
+         * OnResponse will be called when the response from server arrived.
+         *  - When StatusElement is received and it is Success, aData will be nullptr.
+         *  - When command specific response is received, aData will be a valid TLVReader initialized with a struct of command
+         * specify response.
+         *
+         * @param[in] apCommandSender: The command sender object that initialized the command transaction continue to exist after
+         * this call completes.
```

Comment:
OK.  That was not even in the set of possible interpretations of the comment that I considered, so we have clarification work to do here.

That said, we are passing a `const CommandSender *` here.  So it can't be destroyed via the pointer we passed in.  The app _could_ destroy it by matching against other, non-const, state it has, I guess....

If that's what we are trying to avoid, we really need to make this comment clearer, probably with a pointer to the finalization callback as the place where you're allowed to destroy the object.